examples/arkode/F2003_serial/ark_bruss1D_FEM_klu_f2003.f90

```diff
@@ -0,0 +1,1431 @@
+   double precision, parameter :: a = 0.6d0         ! constant forcing on u
+   double precision, parameter :: b = 2.d0          ! steady-state value of w
+   double precision, parameter :: du = 2.5d-2       ! diffusion coeff for u
+   double precision, parameter :: dv = 2.5d-2       ! diffusion coeff for v
+   double precision, parameter :: dw = 2.5d-2       ! diffusion coeff for w
+   double precision, parameter :: ep = 1.d-5        ! stiffness parameter
+   double precision, dimension(N) :: x              ! mesh node locations
+
+ contains
+
+   ! function that maps 2D data into 1D address space
+   ! (0-based since CSR matrix will be sent to C solver)
+   integer(c_int) function idx(ix, ivar)
+     integer(c_int) :: ivar, ix
+     idx = neqreal*(ix - 1) + ivar - 1
+   end function idx
+
+end module UserData
+
+! finite element basis functions
+module FEM
+
+contains
+
+   ! left/right basis functions
+   double precision function ChiL(xl,xr,x)
+      double precision :: xl, xr, x
+      ChiL = (xr-x)/(xr-xl)
+   end function ChiL
+
+   double precision function ChiR(xl,xr,x)
+      double precision :: xl, xr, x
+      ChiR = (x-xl)/(xr-xl)
+   end function ChiR
+
+   ! derivatives of left/right basis functions
+   double precision function ChiL_x(xl,xr)
+      double precision :: xl, xr
+      ChiL_x = 1.d0/(xl-xr)
+   end function ChiL_X
+
+   double precision function ChiR_x(xl,xr)
+      double precision :: xl, xr
+      ChiR_x = 1.d0/(xr-xl)
+   end function ChiR_x
+
+   ! FEM output evaluation routines: value and derivative
+   double precision function Eval(ul,ur,xl,xr,x)
+      double precision :: ul, ur, xl, xr, x
+      Eval = ul*ChiL(xl,xr,x) + ur*ChiR(xl,xr,x)
+   end function Eval
+
+   double precision function Eval_x(ul,ur,xl,xr)
+      double precision :: ul, ur, xl, xr
+      Eval_x = ul*ChiL_x(xl,xr) + ur*ChiR_x(xl,xr)
+   end function Eval_x
+
+end module FEM
+
+! quadrature data
+module Quadrature
+
+contains
+
+  ! nodes
+  double precision function X1(xl,xr)
+     double precision :: xl, xr
+     X1 = 0.5d0*(xl+xr) - 0.5d0*(xr-xl)*0.774596669241483377035853079956d0
+  end function X1
+
+  double precision function X2(xl,xr)
+     double precision :: xl, xr
+     X2 = 0.5d0*(xl+xr)
+  end function X2
+
+  double precision function X3(xl,xr)
+     double precision :: xl, xr
+     X3 = 0.5d0*(xl+xr) + 0.5d0*(xr-xl)*0.774596669241483377035853079956d0
+  end function X3
+
+  ! quadrature
+  double precision function Quad(f1,f2,f3,xl,xr)
+    ! Update these to use "double precision" instead of double precision
+    double precision :: f1, f2, f3, xl, xr
+    double precision, parameter :: wt1=0.55555555555555555555555555555556d0
+    double precision, parameter :: wt2=0.88888888888888888888888888888889d0
+    double precision, parameter :: wt3=0.55555555555555555555555555555556d0
+    Quad = 0.5d0*(xr-xl)*(wt1*f1 + wt2*f2 + wt3*f3)
+  end function Quad
+
+end module Quadrature
+
+module ode_mod
```

Comment:
Please update this module name to be more "unique" (to avoid compilation issues with future/other examples that use overly-generic Fortran module names).